### DISCUSSION TITLE
Is it possible to show an Avalonia window in a WPF application?

### DISCUSSION THREAD

**Discussion by pieranto** [AvaloniaUI/Avalonia#7193] [Q&A] [answered]
We are developing a cross-platform desktop application using Avalonia, but we have also the need to eventually open its main window as a part of an already existing WPF application.

Is it possible? If so, how can we achieve this result?

We have already created a class library project containing the Avalonia window and we have then referenced it in a WPF project without success.

Launch it as an external program is not an option for our use case.

**Comment by Gillibald** [comment]
Have a look at https://github.com/AvaloniaUI/Avalonia/tree/master/samples/interop/WindowsInteropTest

**Reply by pieranto** [reply]
Thank you very much for your reply, I had a look to the interoperability samples and I tried to adapt them to our needs. The problem I have encountered is that our WPF application is realized using .NET 5.0 and the Avalonia.Win32.Interop package seems no to be compatible with .NET 5.0. Am I wrong? Is there any other way to open an Avalonia window (or user control) in a .NET 5.0 WPF Application?

**Reply by Gillibald** [reply]
`Avalonia.Win32.Interop` targets `net461` we should probably multitarget it

**Reply by maxkatz6** [reply]
As an alternative to this project you can use [that](https://github.com/maxkatz6/AvaloniaHwndHostSample)

**Comment by maxkatz6** [comment, marked as answer]
I am pretty sure it's possible to just open avalonia window. Don't forget to run application with builder as usual.

**Reply by pieranto** [reply]
Thank you very much, the link you provided me in you previous reply
> As an alternative to this project you can use [that](https://github.com/maxkatz6/AvaloniaHwndHostSample)

helped us to better understand how to call the builder out of a non Avalonia application, so finally we were able to open our Avalonia window in our WPF application!
